packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1415,7 +1415,8 @@ export class Domain extends DomainBase implements IDomain {
       // Use a custom resource to set the log group resource policy since it is not supported by CDK and cfn.
       // https://github.com/aws/aws-cdk/issues/5343
       logGroupResourcePolicy = new LogGroupResourcePolicy(this, 'ESLogGroupPolicy', {
-        policyName: 'ESLogPolicy',
+        // create a cloudwatch logs resource policy name that is unique to this domain instance
+        policyName: `ESLogPolicy${this.node.addr}`,
```

Comment:
We should also use this is the logical id of the custom resource, otherwise the previous policy would remain orphaned. This would be the behavior if CFN had support for log policies and would generate a physical name for us.